src/odemis/driver/test/symphotime_test.py

```diff
@@ -0,0 +1,133 @@
+    def setUpClass(self):
+        # note: The symphotime simulator must be running before we start the tests. 
+        self.controller = symphotime.Controller(**CONFIG_SYMPHOTIME_SIM)
+        self.controller.updateMetadata(MD)
+
+    @classmethod
+    def tearDownClass(self):
+        self.controller.terminate()
+    
+    def test_acquisition(self):
+        # Test the starting and stopping of acquisition. 
+        
+        for filename in ["file1.ptu", "big$file.ptu"]:
+            for directory in ["MyMeasurement", "TestGroup"]:
+                self.controller.scanner.filename.value = filename
+                self.controller.scanner.directory.value = directory
+
+                # This function will execute once the measurement completes.
+                def callback(flow, array):
+                    self.assertEqual(array, [[0]])
+                    self.assertEqual(self.controller.scanner.filename.value, filename)
+                    self.assertEqual(self.controller.scanner.directory.value, directory)
+                    self.assertEqual(array.metadata[model.MD_DESCRIPTION], MD[model.MD_DESCRIPTION])
+                    self.assertEqual(array.metadata[model.MD_PIXEL_SIZE], MD[model.MD_PIXEL_SIZE])
+                    self.assertEqual(self.controller.scanner.dwellTime.value, array.metadata[model.MD_DWELL_TIME])
+                    # unsub
+                    self.controller.data.unsubscribe(callback)
+                    self.assertFalse(self.controller.isMeasuring())
+
+                # Start measuring
+                self.controller.data.subscribe(callback)
+
+                # Should start measuring now
+                self.assertTrue(self.controller.isMeasuring())
+
+                # Wait until the measurement completes
+                self.controller.waitTillMeasurementComplete()
+                self.assertFalse(self.controller.isMeasuring())
+
+    def test_bad_filename(self):
+        # Test if the filename will get an extension added.
+        filename = "filename"
+        self.controller.scanner.filename.value = filename
+        self.assertEqual(self.controller.scanner.filename.value, filename + '.ptu')
+
+        # Test that a filename that is to long is rejected.
+        with self.assertRaises(ValueError):
+            self.controller.scanner.filename.value = 'a' * 256
+
+    def test_stopping(self):
+        # Start measuring, then immediately stop. Be sure that the measurement takes at least 5 seconds.
+        
+        # This function will execute once the measurement completes.
+        def callback(flow, array):
+            self.fail("This is not supposed to run, since measurement is supposed to stop. ")
+
+        # Start measuring, wait, then stop
+        self.controller.data.subscribe(callback)
+        self.assertTrue(self.controller.isMeasuring())
+        self.controller.data.unsubscribe(callback)
+        # Measurement should be stopped
+        self.controller.waitTillMeasurementComplete()
```

Comment:
Don't use `waitTillMeasurementComplete()`, just wait a little bit, and check it's not measuring anymore.